plugins/cilium-cni/interface.go

```diff
@@ -38,6 +38,8 @@ func interfaceAdd(ipConfig *current.IPConfig, ipam *models.IPAMAddressResponse,
 	ipv4CIDRs, _ := ip.CoalesceCIDRs(allCIDRs)
 	cidrs := make([]net.IPNet, 0, len(ipv4CIDRs))
 	for _, cidr := range ipv4CIDRs {
+		// TODO check with reviewer if any mistake on this as cidr slices is not used now
```

Comment:
Just want to confirm cidrs slice should be used in `enirouting.NewRoutingInfo` at line 46.

Reply:
Hmm this lint message is deceiving. The linter is correct in that the `cidrs` slice should be used, but then that wouldn't compile because of type issues with `NewRoutingInfo`.

The crux of the issue is that `NewRoutingInfo` takes in as the second parameter, a slice of strings representing the CIDRs. This code here is taking `ipam.Cidrs` and validating it (`NewRoutingInfo` already does this, but that's beside the point), and then coalescing it into `ipv4CIDRs`. Then the next for-loop creates a `[]net.IPNet` which is different than `ipv4CIDRs` because it's a slice of `*net.IPNet`. 

The simplest thing to do here is have the for-loop for `cidrs` convert each element back into the string form, and pass that to `NewRoutingInfo`. 

There's a lot of duplicated and confusing logic here :confused:

Reply:
@christarazi thanks for your comment, it's exactly what I think is right. But I am not fully confident :).

Reply:
let me address this one in separate PR if you don't mind :)

Reply:
@sayboras Go for it!

Reply:
https://github.com/cilium/cilium/issues/11562 raised for tracking purpose

Reply:
Now that this is being tracked, we can remove this TODO comment.